src/newlines.cpp

```diff
@@ -3231,7 +3231,7 @@ void newlines_chunk_pos(c_token_t chunk_type, tokenpos_e mode)
    chunk_t *prev;
    int     nl_flag;
 
-   if ((mode & (TP_JOIN | TP_LEAD | TP_TRAIL)) == 0)
+   if ((mode & (TP_JOIN | TP_LEAD | TP_TRAIL)) == 0 && chunk_type != CT_COMMA)
```

Comment:
What was the reasoning behind this change ? This changes behaviour and it's not entirely related to the new option. Basically you're saying that now all modes are valid unless chunk_type is not a comma. Can you show with a test the effect of what this does and how it was wrong previously for `CT_CLASS_BASE` ?